### PR TITLE
New API for translating UTF-8 to UV

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

File: md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+They differ in how they handle input which results in a <false> return.
+
+More precisely, they each calculate the first code point represented by the
+sequence of bytes bounded by <*s> .. <(*e) - 1>, interpreted as UTF-8.
+<e> must be strictly greather than <s>; this is asserted for in debugging
+builds.
+
+Since UTF-8 is a variable length encoding, the number of bytes examined also
+varies.  The algorithm is to first look at <*s>.  If that represents a full
+UTF-8 character, no other byte is examined, and the UTF-8 is calculated from
```
Comment (tonycoz) [inline md:49]: "and the UTF-8 is calculated"

should this be:

"and the code point is calculated"
?

File: md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+     }
+
+     s += len;
+ }
+
+C<utf8_to_uvchr> will raise warnings for malformations if UTF8 warnings are
+enabled;  C<utf8_to_uvchr_nowarn> will never raise a warning.
+
+Perl extends Unicode to include extra code points.  Neither C<utf8_to_uvchr>
+nor C<utf8_to_uvchr_nowarn> will raise warnings for these.
```
Comment (tonycoz) [inline md:106]: Does this include surrogates?

File: md (hunk 1 above)
Comment (khwilliamson) [reply md]: Yes, thanks

File: md (hunk 2 above)
Comment (khwilliamson) [reply md]: see revision

**Comment by book** [comment]
@khwilliamson Is this PPC proposal still relevant?

**Comment by book** [comment]
Closing, as suggested by @khwilliamson on IRC.

**Pull request closed without merging**